app/celery/process_ses_receipts_tasks.py

```diff
@@ -18,6 +19,77 @@
 from celery.exceptions import Retry
 
 
+def handle_complaints_and_extract_ref_ids(messages):
+    """Processes the current batch of notification receipts. Handles complaints, removing them from the batch
+       and returning the remaining messages for further processing.
+
+    Args:
+        messages (List): List of SES messages received from the SQS receipt buffer queue.
+
+    Returns:
+        Tuple: A tuple containing a list of notification reference IDs and a reduced list of SES messages not containing any complaint receipts.
+    """
+    ref_ids = []
+    complaint_free_messages = []
+
+    for message in messages:
+        notification_type = message["notificationType"]
+        if notification_type == "Complaint":
+            _check_and_queue_complaint_callback_task(*handle_complaint(message))
+        else:
+            ref_ids.append(message["mail"]["messageId"])
+            complaint_free_messages.append(message)
+
+    return ref_ids, complaint_free_messages
+
+
+def prepare_updates_and_retries(ses_messages, notifications):
+    """Prepares a list of updates and retries for the notifications based on the SES receipts received.
+
+    Args:
+        ses_messages (List): The SES receipts received from the SQS queue. Should not contain any complaint receipts.
```

Comment:
I am looking at the diagram and the code .. I cant see where the complaints will be referenced if not in this func?

Reply:
The method above this one, `handle_complaints_and_extract_ref_ids`, is called at the beginning of `process_ses_results`.  Similar to the [previous code](https://github.com/cds-snc/notification-api/blob/fdb9ffed0e5b2e413deb69af7b0a7dbc7c78a3cd/app/celery/process_ses_receipts_tasks.py#L45) it passes complaint receipts to [`handle_complaint`](https://github.com/cds-snc/notification-api/blob/main/app/notifications/notifications_ses_callback.py#L146) which is a separate process that adds those notifications to our `complaint` table.